### ISSUE TITLE
Allow for file Attachments

### ISSUE THREAD

**Issue by scabana** [xunit/xunit#936]
We are migrating from MsTest to xunit and one thing that is missing from xunit are file attachments. MsTest allows your to add files to the result using `TestContext.AddResultFile(fileName)`. The place where to push those files (for vstest.console.exe) is in the `Attachments` property of the `Microsoft.VisualStudio.TestPlatform.ObjectModel.TestResult` class. Our usecase is to attach screenshots on failure or dynamicaly created data for some tests.

**Comment by SDETRK** [comment]
This is definitely needed

**Comment by jordanchang** [comment]
I'm working on a Selenium test project for a website and would really like to attach a screenshot to failed tests.

**Comment by sandersaares** [comment]
I would also like not only attachments but writable file streams - my tests (as a whole, not individually) generate tracing data very useful for failure analysis. However, I am currently left to handle it completely outside the xunit universe, in parallel to actual test code. This is suboptimal.

It would be especially great if this worked also for platforms where normal file I/O is otherwise permission-limited (e.g. UWP).

**Comment by scabana** [comment]
The writable streams would be great. We don't have the limitation of platforms like you do, but it would definitely save some work.

**Comment by MaxWilson** [comment]
Is there a workaround for VSTest, or are we just out of luck for now when it comes to screenshots?

**Comment by jordanchang** [comment]
I switched my project to nUnit. At this point, it has everything I need
including running tests in parallel. xunit is cool, but still half baked.

**Comment by bradwilson** [comment]
Closed for age.

**Comment by sandersaares** [comment]
Is this issue no longer relevant?

**Comment by bradwilson** [comment]
https://twitter.com/bradwilson/status/929046059640225793

**Comment by altrive** [comment]
It seems xunit v3 expeced to have "test context" feature.  #621
Is there any plan to support file attachments to test result feature also?

**Comment by SychevIgor** [comment]
It will be nice to have this feature in Xunit.
Our UX testing team choose NUnit only because they can't attach screenshots of failed tests in XUnit.

**Comment by sjoerdvanloon** [comment]
I was really in love with the XUnit framework, but I am really missing this feature and am considering to switch back to NUnit for my company UX tests. Are there still no plans to include this feature?

**Comment by pabrams** [comment]
Why's this closed? Does that mean it won't be considered as a feature?

**Comment by filzrev** [comment]
There is following feature ticket on xUnit.net v3 roadmap.
https://github.com/xunit/xunit/issues/2457

If this feature will be supported by v3.
Attached files can display files on VS Test Explorer.
